url2a908001cd_malictus_klang_tgz-pJ8-malictus_klang_test_PrimitiveTestJ8/src/malictus/klang/ui/klangeditor/RawEditPanel.java

```diff
@@ -0,0 +1,585 @@
+							this.btnPrimView.setEnabled(true);
+						}
+						PrimitiveData pd = new PrimitiveData(prim, KlangConstants.PRIMITIVE_DATA_RAW_VALUE_NAME, start, end);
+						this.curPrim = pd;
+					}
+					//overwrite of KlangLabel's resetting of label's width
+					lblPrimView2.setPreferredSize(new Dimension(80, lblPrimView2.getHeight()));
+				}
+				lblPrimView2.setEnabled(true);
+			}
+			raf.close();
+		} catch (Exception err) {
+			if (raf != null) {
+				try {
+					raf.close();
+				} catch (Exception foo) {}
+			}
+		}
+	}
+	
+	/**
+	 * Return the ASCII representation of an integer, but only if it's one of regaular ASCII characters
+	 * @param val the integer value
+	 * @return the ASCII value that this integer represents, or a space if the ASCII symbol is out of the normal range
+	 */
+	private String convertASCIIString(int val) {
+		if ((val <= 32) || (val > 126)) {
+			return "&nbsp;";
+		}
+		//this one character isn't displaying correctly in HTML; are there others?
+		if (val == 60) {
+			return "&#60;";
+		}
+		return new Character((char)val).toString();
+	}
+	
+	/**
+	 * Append a zero to front of a hex string if it's a single digit
+	 * @param input the input hex string
+	 * @return the reformatted hex string
+	 */
+	private String convertHexString(String input) {
+		if (input.length() < 2) {
+			input = "0" + input;
+		}
+		return input;
+	}
+	
+	/**
+	 * Move forward or backwards from the current position
+	 * @param amt the amt to move; may be positive or negative
+	 */
+	private void doMove(int amt) {
+		long newbyte = this.parent.currentByte + amt;
+		if (newbyte < 0) {
+			newbyte = 0;
+		}
+		if (newbyte >= this.parent.getCurrentFile().getFile().length()) {
+			newbyte = this.parent.getCurrentFile().getFile().length() - 1;
+		}
+		parent.jumpToByte(newbyte);
+	}
+	
+	/**
+	 * Called whenever the jump button is pressed
+	 */
+	private void doJump() {
+		if (this.txtfCurByte.getText().trim().equals("")) {
+			return;
+		}
+		String x = txtfCurByte.getText().trim();
+		try {
+			Long lo = Long.decode(x);
+			if (lo.longValue() < 0) {
+				//ignore
+				updatePanel();
+			} else {
+				if (lo.longValue() >= parent.getCurrentFile().getFile().length()) {
+					lo = new Long(parent.getCurrentFile().getFile().length() - 1);
+				}
+				parent.jumpToByte(lo.longValue());
+				updatePanel();
+			}
+		} catch (Exception err) {
+			//just ignore and reset
+			updatePanel();
+		}
+	}
+	
+	/**
+	 * Edit the current prim value, as selected in the prim combo box.
+	 */
+	private void doEditCurrentPrim() {
+		//double check
+		if (parent.isReadOnlyMode()) {
+			return;
+		}
+		PrimEditDialog pep = new PrimEditDialog(curPrim, this.parent);
+		if (pep.wasCanceled()) {
+			return;
+		}
+		if (!pep.valueChanged()) {
+			return;
+		}
+		//alter the bytes
+		RandomAccessFile raf = null;
+		try {
+			curPrim.getPrimitive().setValueFromString(pep.getNewValue());
+			raf = new RandomAccessFile(parent.getCurrentFile().getFile(), "rw");
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *raf*. Currently, there are execution paths that do not contain closure statements, for example, when RandomAccessFile.seek throws an exception. Ensure *raf* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).